include/bredis/AsyncConnection.hpp

```diff
@@ -0,0 +1,69 @@
+//
+//
+// Copyright (c) 2017 Ivan Baidakou (basiliscos) (the dot dmol at gmail dot com)
+//
+// Distributed under the MIT Software License
+//
+
+#pragma once
+
+#include <atomic>
+#include <initializer_list>
+#include <mutex>
+#include <queue>
+#include <sstream>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include <boost/asio.hpp>
+#include <boost/utility/string_ref.hpp>
+
+#include "Command.hpp"
+#include "Error.hpp"
+#include "Protocol.hpp"
+#include "Result.hpp"
+
+namespace bredis {
+
+template <typename AsyncStream> class AsyncConnection {
```

Comment:
Consider renaming `AsyncConnection` to just `Connection`
Consider renaming `AsyncStream` to `NextLayer`